soroban-env-host/src/host.rs

```diff
@@ -901,8 +901,9 @@ impl Host {
         })?;
 
         let lk = LedgerKey::Account(LedgerKeyAccount {
-            account_id: AccountId(PublicKey::PublicKeyTypeEd25519(self.to_u256(account_id)?)),
+            account_id: self.to_account_id(account)?,
```

Comment:
Please make `account_id` building consistent with `to_trustline_key`. Also I'm not sure if this one is correct, we want `AccountId` xdr here, don't we?

Reply:
Just to be clear, are you saying I should add a `to_account_key` method? And `to_account_id` returns the xdr `AccountId`. It's a little confusing, but the `AccountId` in base_types is only used in the token contract.

Reply:
If `to_account_id` returns the xdr, then why don't we use it in `to_trustline_key`? We build account ids for both trustline user and issuer manually there.

Reply:
Do we maybe have different inputs to these methods? Maybe we should document what are the expected `Object` types, because the code is pretty hard to follow currently (or maybe use explicit types instead of `Object`s where possible).

Reply:
Yeah that's a good point. I added `to_account_id` after `to_u256_from_account` and missed this. My goal here was to have all of these methods accept `HostObject::AccountId` instead of `HostObject::Bytes`, which is what `to_u256` accepts. I updated `to_trustline_key` to use `account_id`.

I agree we should add documentation about the expected `Object` types. I'll open an issue for this.

Reply:
Thanks! The change LGTM now.